File: app/src/main/java/com/pixplicity/cryptogram/models/PuzzleProgressState.java
package com.pixplicity.cryptogram.models;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class PuzzleProgressState {

    @SerializedName("current")
    private Integer mCurrentPuzzleId;

    @SerializedName("progress")
    private ArrayList<PuzzleProgress> mProgress;

    @Nullable
    public Integer getCurrentId() {
        return mCurrentPuzzleId;
    }

    public void setCurrentId(int puzzleId) {
        mCurrentPuzzleId = puzzleId;
    }

    @NonNull
    public ArrayList<PuzzleProgress> getProgress() {
        if (mProgress == null) {
            mProgress = new ArrayList<>();
        }
        return mProgress;
    }

    public void addProgress(PuzzleProgress puzzleProgress) {
        getProgress().add(puzzleProgress);
    }

}

File: app/src/main/java/com/pixplicity/cryptogram/models/PuzzleProgressState.kt
package com.pixplicity.cryptogram.models

import com.google.gson.annotations.SerializedName

import java.util.ArrayList

class PuzzleProgressState {

    @SerializedName("current")
    var currentId: Int? = null
        private set

    @SerializedName("progress")
    private var mProgress: ArrayList<PuzzleProgress>? = null

    val progress: ArrayList<PuzzleProgress>
        get() {
            if (mProgress == null) {
                mProgress = ArrayList()
            }
            return mProgress!!
        }

    fun setCurrentId(puzzleId: Int) {
        currentId = puzzleId
    }

    fun addProgress(puzzleProgress: PuzzleProgress) {
        progress.add(puzzleProgress)
    }

}
